examples/datetime-timezone/app.py

```diff
@@ -0,0 +1,119 @@
+from datetime import datetime
+from zoneinfo import ZoneInfo
+
+from flask import send_from_directory, Flask, request, session, jsonify
+from flask_sqlalchemy_lite import SQLAlchemy
+from markupsafe import Markup
+from sqlalchemy import DateTime, String
+from sqlalchemy.orm import DeclarativeBase, Mapped, mapped_column
+
+from flask_admin import Admin
+from flask_admin.contrib.sqla import ModelView
+from flask_admin.model import typefmt
+
+
+# model
+class Base(DeclarativeBase):
+    pass
+
+
+class Article(Base):
+    __tablename__ = "article"
+    id: Mapped[int] = mapped_column(primary_key=True)
+    text: Mapped[str] = mapped_column(String(30))
+    last_edit: Mapped[datetime] = mapped_column(DateTime(timezone=True))
+
+
+# app
+app = Flask(__name__)
+app.config["SQLALCHEMY_ENGINES"] = {"default": "sqlite:///default.sqlite"}
+# Create dummy secret key so we can use sessions
+app.config['SECRET_KEY'] = '123456789'
+db = SQLAlchemy(app)
+
+
+# admin
+def date_format(view, value):
+    """
+    Ensure consistent date format and inject class for timezone.js parser.
+    """
+    if value is None:
+        return ''
+    return Markup(
+        f'<span class="timezone-aware">{value.strftime("%Y-%m-%d %H:%M:%S")}</span>'
+    )
+
+
+MY_DEFAULT_FORMATTERS = dict(typefmt.BASE_FORMATTERS)
+MY_DEFAULT_FORMATTERS.update({
+    datetime: date_format,
+})
+
+
+class TimezoneAwareModelView(ModelView):
+    column_type_formatters = MY_DEFAULT_FORMATTERS
+    extra_js = ['/timezone.js']
+
+    def on_model_change(self, form, model, is_created):
+        """
+
+        """
+        user_timezone = session["timezone"]
+
+        for field_name, field_value in form.data.items():
+            if isinstance(field_value, datetime):
+                # Convert naive datetime to timezone-aware datetime
+                aware_time = field_value.replace(tzinfo=ZoneInfo(user_timezone))
+
+                # Convert the time to UTC
+                utc_time = aware_time.astimezone(ZoneInfo('UTC'))
+
+                # Assign the UTC time to the model
+                setattr(model, field_name, utc_time)
+
+        super(TimezoneAwareModelView, self).on_model_change(form, model, is_created)
+
+
+# inherit TimeZoneAwareModelView to make any admin page timezone-aware
+class BlogModelView(TimezoneAwareModelView):
+    # any additional code here
+    pass
+
+
+# Flask views
+@app.route('/')
+def index():
+    return '<a href="/admin/article">Click me to get to Admin!</a>'
+
+
+@app.route('/timezone.js/')
+def timezone_js():
+    return send_from_directory('static/js', 'timezone.js')
```

Comment:
We shouldn't need a custom handler for this if we configure Flask correctly. It should be something like changing the Flask instance creation to `app = Flask(__name__, static_folder='static', static_url_path='/static')` and then changing the `extra_js` line to `extra_js = ['/static/js/timezone.js']`

Reply:
You are correct: we can remove the route and changing the extra_js line to `extra_js = ['/static/js/timezone.js']`. I have also leveraged moment.js to remove some code from /static/js/timezone.js

Reply:
I think you've missed updating Flask - the example is broken now

Reply:
are you running it locally? What error do you get?

Reply:
My mistake - didn't spot that you'd added a separate TimezoneAware Article tab now.

Reply:
Yes, I was trying to make it more clear. Maybe I should change the link at / to point to the TimezoneAware tab? Or should I simply drop the regular naive timezone one?

Reply:
I think it's fine to have separate tabs 👍  Maybe the column name for the TimezoneAware tab could be "Last edit (local time)" - or could display the timezone code?

Reply:
I have added (UTC) and (local time) to the labels in https://github.com/pallets-eco/flask-admin/pull/2502/commits/1439622aae142f51a14f81f567bc32826fe5d476 (I have also changed the url to follow snake case: now it's http://127.0.0.1:5000/admin/timezone_aware_article/ and not http://127.0.0.1:5000/admin/timezoneaware_article/). I have chosen to display "(local time)" because it's not immediate to show the timezone code as session["timezone"] might not be set yet when the user visits the page, so we would need to change the label with js.

Reply:
Looks good 👍